rali/rali/source/caffe2_meta_data_reader_detection.cpp

```diff
@@ -0,0 +1,229 @@
+        auto it = _map_content.find(image_name);
+        if (_map_content.end() == it)
+            THROW("ERROR: Given name not present in the map" + image_name)
+        _output->get_bb_cords_batch()[i] = it->second->get_bb_cords();
+        _output->get_bb_labels_batch()[i] = it->second->get_bb_labels();
+    }
+}
+
+void Caffe2MetaDataReaderDetection::print_map_contents()
+{
+    BoundingBoxCords bb_coords;
+    BoundingBoxLabels bb_labels;
+
+    std::cerr << "\nMap contents: \n";
+    for (auto &elem : _map_content)
+    {
+        std::cerr << "Name :\t " << elem.first;
+        bb_coords = elem.second->get_bb_cords();
+        bb_labels = elem.second->get_bb_labels();
+        std::cerr << "\nsize of the element  : " << bb_coords.size() << std::endl;
+        for (unsigned int i = 0; i < bb_coords.size(); i++)
+        {
+            std::cerr << " x : " << bb_coords[i].x << " y: :" << bb_coords[i].y << " width : " << bb_coords[i].w << " height: :" << bb_coords[i].h << std::endl;
+            std::cerr << "Label Id : " << bb_labels[i] << std::endl;
+        }
+    }
+}
+
+void Caffe2MetaDataReaderDetection::read_all(const std::string &path)
+{
+    string tmp1 = path + "/data.mdb";
+    string tmp2 = path + "/lock.mdb";
+    uint file_size, file_size1, file_bytes;
+
+    ifstream in_file(tmp1, ios::binary);
+    in_file.seekg(0, ios::end);
+    file_size = in_file.tellg();
+    ifstream in_file1(tmp2, ios::binary);
+    in_file1.seekg(0, ios::end);
+    file_size1 = in_file1.tellg();
+    file_bytes = file_size + file_size1;
+    read_lmdb_record(path, file_bytes);
+    // print_map_contents();
+}
+
+void Caffe2MetaDataReaderDetection::read_lmdb_record(std::string file_name, uint file_byte_size)
+{
+    int rc;
+    MDB_env *env;
+    MDB_dbi dbi;
+    MDB_val key, data;
+    MDB_txn *txn;
+    MDB_cursor *cursor;
+    string str_key, str_data;
+
+    // Creating an LMDB environment handle
+    E(mdb_env_create(&env));
+    // Setting the size of the memory map to use for this environment.
+    // The size of the memory map is also the maximum size of the database.
+    E(mdb_env_set_mapsize(env, file_byte_size));
+    // Opening an environment handle.
+    E(mdb_env_open(env, file_name.c_str(), 0, 0664));
+    // Creating a transaction for use with the environment.
+    E(mdb_txn_begin(env, NULL, MDB_RDONLY, &txn));
+    // Opening a database in the environment.
+    E(mdb_dbi_open(txn, NULL, 0, &dbi));
+
+    // Creating a cursor handle.
+    // A cursor is associated with a specific transaction and database
+    E(mdb_cursor_open(txn, dbi, &cursor));
+
+    // Retrieve by cursor. It retrieves key/data pairs from the database
+    while ((rc = mdb_cursor_get(cursor, &key, &data, MDB_NEXT)) == 0)
+    {
+        // Reading the key value for each record from LMDB
+        str_key = string((char *)key.mv_data);
+        // Reading the data value for each record from LMDB
+        str_data = string((char *)data.mv_data);
```

Comment:
Codacy found an issue: [Variable 'str_data' is assigned a value that is never used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5976200)